prlworkflows/sqs.py

```diff
@@ -92,20 +92,21 @@ def _subl_error():
         self_copy.replace_species(replacement_dict)
 
         if scale_volume:
-            if len(self_copy) == 1:
-                # we only have one element, just set the scaling factor to scale to the pure element density
-                sf = self_copy.density / (float(self_copy.sites[0].specie.data['Density of solid'].split(' ')[0])/1e3)
-            else:
-                idx = np.nonzero(self_copy.distance_matrix)  # exclude distance from self
-                # construct a minimal distance matrix based on average ionic radii
-                radius_matrix = np.zeros((len(self_copy.sites), len(self_copy.sites)))
-                for i in range(len(self_copy.sites)):
-                    for j in range(len(self_copy.sites)):
-                        radius_matrix[i, j] = (
-                            self_copy.sites[i].specie.data['Atomic radius'] + self_copy.sites[j].specie.data['Atomic radius'])
-                # find the scale factor and scale the lattice
-                sf = np.max(radius_matrix[idx] / self_copy.distance_matrix[idx]) ** 3
-            self_copy.scale_lattice(self_copy.volume * sf)
+            fractional_comp = dict(self_copy.composition.fractional_composition)
+            f = []
+            for key in fractional_comp:
+                f.append(fractional_comp[key])
```

Comment:
This method will calculate incorrect densities. The dictionary of fractional compositions is not sorted. 

Imagine you have a two component system, Fe-Ni. On different runs, your `f` array might have the density of Fe first and other times it might have the density of Ni first. You then proceed through the elements in the order you get them from `self_copy.composition.elements`, which may not have the same order as your density array. 

The solution is to either 

1. make sure your `f` array is sorted in the same way as `self_copy.composition.elements`. There are ways to do this, but they are not very clear, IMO.
2. (preferred) Use the dictionary you created (`fractional_comp `) to look up elements by string name in the loop. You don't need to create the `f` array at all. Then you also don't have to use a counter in the loop because you are accessing by name, not index.